docs/source/design/ui_design.rst

```diff
@@ -255,69 +255,85 @@ dictionary with a schema defined by QCElemental.
-``calculate_X_wavefunction``. It should be noted that, caching mechanism of
-NWChemEx avoids any recomputation if the requested quantity is already calculated.
-
-Our UI also provides a more general ``nwx.calculate`` function, where the method
-and the return types can be specified in the ``options`` explicitly or provided
-as key-value arguments in the function call.
+    import nwchemex as nwx 
+    energy = nwx.calculate(molecule = 'H 0. 0. 0. \n H 0., 0. 1.', method='scf', basis = 'sto-3g')
+
+While this function can be enough for most common tasks, the flexibility of this
+function is limited particularly for the additional outputs that a user might be
+interested in, such as the wavefunction, density matrix, or intermediate outputs
+such as the data related to the convergence or timings of the SCF iterations. To
+address this issue, we provide a more flexible ``run()`` function with the
+following signature.
 
 .. code-block:: python
 
-    def calculate(molecule: Union[str, chemist.Molecule], method: str, basis:
-    Union[str, simde.type.ao_space], return_energy: Bool = True, options:
-    Union[Dict, dataclass] = None, **kwargs):
+    def run(simulation: nwx.Simulation, **kwargs) -> nwx.Simulation:
 
-Using this function, a user can run any method and request different return
-types either using ``options`` or by setting these arguments explicitly.
+In this function, the complexity of all the input arguments and return types is
+buried under the ``nwx.Simulation`` object. The user can create a
+``nwx.Simulation`` object and set the required attributes and then pass this
+object to the ``run()`` function or use this function similar to the
+``calculate()`` function by using keyword arguments. The return type of the
+``run()`` function is also a ``nwx.Simulation`` object, which contains all the
+input provided by the user and all the outputs generated by the calculation.
 
 .. code-block:: python
-    
-    # Set arguments using options
+
     import nwchemex as nwx
-    options = nwx.options(method = 'scf', basis = 'sto-3g', return_gradients=True)
-    energy, gradients = nwx.calculate(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', options = options)
-    
-    # Set arguments explicitly
+    sim = nwx.Simulation()
+    sim.molecule = 'H 0. 0. 0. \n H 0. 0. 1.'
+    sim.method = 'scf'
+    sim.basis = 'sto-3g'
+    sim.return_gradients = True
+    sim.return_wavefunction = True
+    sim = nwx.run(sim)
+    energy = sim.energy
+    gradients = sim.gradients
+    wavefunction = sim.wavefunction
+
+Alternatively with keyword arguments:
+.. code-block:: python
+
     import nwchemex as nwx 
-    energy, gradients = nwx.calculate(molecule = 'H 0. 0. 0. \n H 0. 0. 1.',
-    method = 'scf', basis = 'sto-3g', return_gradients=True)
+    sim = nwx.run(nwx.Simulation(), molecule = 'H 0. 0. 0. \n H 0. 0. 1.',
+    method='scf', basis = 'sto-3g', return_gradients = True, return_wavefunction
+    = True)
```

Comment:
```suggestion
The full signature of `calculate` is:

.. code-block:: python

   def calculate(mol = Union[str, nwx.Molecule], method = str, basis = Union[str, nwx.AOBasisSet, nwx.AOSpace], options = Options, mpi_comm = None, module_manager = None)
```

`Options` is an opaque type (to be designed later in coordination with PluginPlay #308) which is capable of holding key/value pairs for inputs, as well as a list of results the user wants to compute.